llama_stack/providers/inline/post_training/common/utils.py

```diff
@@ -0,0 +1,45 @@
+
+import torch
+import pynvml
+from typing import List
+
+from pydantic import BaseModel
+
+class GPUInformation(BaseModel):
+    utilization: int
+    power_draw: float
+    total_memory: int
+    used_memory: int
+
+class SystemInformation:
+    def __init__(
+        self
+    ) -> None:
+        """Detect available GPU architecture and get memory usage."""
+        gpus = None
+        if torch.cuda.is_available():
+           gpus = self.get_cuda_usage()
+        else:
+            return ["No supported GPU detected."]
+        self.gpu_information = gpus
+
+
+    def get_cuda_usage(self) -> tuple[list, int, int, int, int]:
+        """Get GPU memory usage for NVIDIA GPUs using NVML."""
+        try:
+            from pynvml import nvmlInit, nvmlDeviceGetHandleByIndex, nvmlDeviceGetMemoryInfo, nvmlDeviceGetUtilizationRates, nvmlDeviceGetPowerUsage, nvmlShutdown
```

Comment:
Wrap import in try/catch but put logic outside of it- that'll prevent erroneous error catching